packages/manager/modules/vrack/src/vrack.routing.js

```diff
@@ -24,6 +25,22 @@ export default /* @ngInject */ ($stateProvider) => {
       staticResources: () => true,
       defaultFilterColumn: () => 'serviceName',
       header: /* @ngInject */ ($translate) => $translate.instant('vrack_title'),
+      surveyUrl: /* @ngInject */ (coreConfig) => {
+        // Get default language
+        const defaultLanguage = Object.keys(ALLOWED_LANGUAGES).find(
+          (key) => ALLOWED_LANGUAGES[key].isDefault,
+        );
+        const userLanguage = coreConfig.getUserLanguage();
+        const languageToUse =
+          typeof ALLOWED_LANGUAGES[userLanguage] === 'object'
+            ? userLanguage
+            : defaultLanguage;
+        // Get user
+        const user = coreConfig.getUser();
+
+        // Build url for survey link
+        return `${BASE_URL_SURVEY}${languageToUse}&nic=${user.nichandle}`;
```

Comment:
This code is already defined, try to put it in a service exposed by your module and use it in the other locations